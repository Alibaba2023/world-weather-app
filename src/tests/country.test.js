import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Country from '../components/country';

const countryData = {
  name: 'Sample Country',
  flag: 'https://example.com/sample-flag.png',
};

describe('Country component', () => {
  it('renders the country and comparing to Snapshot', () => {
    const countryComponent = render(
      <MemoryRouter>
        <Country name={countryData.name} flag={countryData.flag} onClick={() => { }} />
      </MemoryRouter>,
    );
    expect(countryComponent).toMatchSnapshot();
  });

  it('triggers onClick event when NavLink is clicked', () => {
    const onClickMock = jest.fn();

    render(
      <MemoryRouter>
        <Country name={countryData.name} flag={countryData.flag} onClick={onClickMock} />
      </MemoryRouter>,
    );

    const navLinkElement = screen.getByRole('link');
    fireEvent.click(navLinkElement);
    expect(onClickMock).toHaveBeenCalledWith(countryData.name);
  });

  it('renders the country name and flag image', () => {
    const countryComponent = render(
      <MemoryRouter>
        <Country name={countryData.name} flag={countryData.flag} onClick={() => { }} />
      </MemoryRouter>,
    );
    expect(countryComponent).toMatchSnapshot();
  });
});
