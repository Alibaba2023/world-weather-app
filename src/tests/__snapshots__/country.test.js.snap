// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country component renders the country and comparing to Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <a
          class=""
          href="/details"
        >
          <div
            class="country-card"
          >
            <img
              alt="flag"
              class="flag-image"
              src="https://example.com/sample-flag.png"
            />
            <h3
              class="country-name"
            >
              Name:
              Sample Country
            </h3>
          </div>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <a
        class=""
        href="/details"
      >
        <div
          class="country-card"
        >
          <img
            alt="flag"
            class="flag-image"
            src="https://example.com/sample-flag.png"
          />
          <h3
            class="country-name"
          >
            Name:
            Sample Country
          </h3>
        </div>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Country component renders the country name and flag image 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <a
          class=""
          href="/details"
        >
          <div
            class="country-card"
          >
            <img
              alt="flag"
              class="flag-image"
              src="https://example.com/sample-flag.png"
            />
            <h3
              class="country-name"
            >
              Name:
              Sample Country
            </h3>
          </div>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <a
        class=""
        href="/details"
      >
        <div
          class="country-card"
        >
          <img
            alt="flag"
            class="flag-image"
            src="https://example.com/sample-flag.png"
          />
          <h3
            class="country-name"
          >
            Name:
            Sample Country
          </h3>
        </div>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
